tests/imputers/test_NearestMeanResponseImputer.py

```diff
@@ -160,161 +158,12 @@ def test_learnt_values2(self):
         )
 
 
-class TestTransform:
-    """Tests for NearestMeanResponseImputer.transform."""
-
-    def expected_df_1():
-        """Expected output for ."""
-        df = pd.DataFrame(
-            {"a": [1, 1, 2, 3, 3, 2], "b": [3, 3, 1, 3, 3, 4], "c": [2, 3, 2, 1, 4, 1]},
-        )
+class TestTransform(
+    GenericTransformTests,
+    GenericImputerTransformTests,
+):
+    """Tests for transformer.transform."""
 
-        df[["a", "b"]] = df[["a", "b"]].astype("float64")
-
-        return df
-
-    def expected_df_2():
-        """Expected output for ."""
-        df = pd.DataFrame(
-            {
-                "a": [1, 1, 2, 3, 3, 2],
-                "b": [np.nan, np.nan, 1, 3, 3, 4],
-                "c": [2, 3, 2, 1, 4, 1],
-            },
-        )
-
-        df["a"] = df["a"].astype("float64")
-
-        return df
-
-    def expected_df_3():
-        """Expected output for ."""
-        df = pd.DataFrame({"a": [2, 3, 4, 1, 4, 2]})
-
-        df["a"] = df["a"].astype("float64")
-
-        return df
-
-    def test_check_is_fitted_called(self, mocker):
-        """Test that BaseTransformer check_is_fitted called."""
-        df = create_NearestMeanResponseImputer_test_df()
-
-        x = NearestMeanResponseImputer(columns=["a", "b"])
-
-        x.fit(df, df["c"])
-
-        expected_call_args = {0: {"args": (["impute_values_"],), "kwargs": {}}}
-
-        with ta.functions.assert_function_call(
-            mocker,
-            tubular.base.BaseTransformer,
-            "check_is_fitted",
-            expected_call_args,
-        ):
-            x.transform(df)
-
-    def test_super_transform_called(self, mocker):
-        """Test that BaseTransformer.transform called."""
-        df = create_NearestMeanResponseImputer_test_df()
-
-        x = NearestMeanResponseImputer(columns=["a", "b"])
-
-        x.fit(df, df["c"])
-
-        expected_call_args = {
-            0: {"args": (create_NearestMeanResponseImputer_test_df(),), "kwargs": {}},
-        }
-
-        with ta.functions.assert_function_call(
-            mocker,
-            tubular.base.BaseTransformer,
-            "transform",
-            expected_call_args,
-        ):
-            x.transform(df)
-
-    @pytest.mark.parametrize(
-        ("df", "expected"),
-        ta.pandas.adjusted_dataframe_params(
-            create_NearestMeanResponseImputer_test_df(),
-            expected_df_1(),
-        ),
-    )
-    def test_nulls_imputed_correctly(self, df, expected):
```

Comment:
Q: do we not need this one? (same Q for the 2 version too)

Reply:
and 3

Reply:
Mighta been a bit hasty to get rid of these (especially 2/3) as thought they were covered by the generic tests. Can put those back in, but isn't the 'test_nulls_imputed_correctly' test in the test_NearestMeanResponseImputer script not covered by the 'test_expected_output_1' test in the 'test_BaseImputer' script?

Reply:
ahh sorry think you were right, they look to be adding anything above the inherited tests. happy to leave removed